### PR TITLE
Adds EKSConfig status conditions

### PR DESCRIPTION
**Pull request by NicoleYson**
Adds status conditions

Notes:
- Did not add anything that had to deal with certs, e.g. `CertificatesAvailableCondition`

### CODE REVIEW

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 1]
```diff
@@ -118,13 +119,19 @@ func (r *EKSConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr e
...
+			),
+			conditions.WithStepCounter(),
+		)
 		patchOpts := []patch.Option{}
 		if rerr == nil {
 			patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
 		}
 		if err := patchHelper.Patch(ctx, config, patchOpts...); err != nil {
 			log.Error(rerr, "Failed to patch config")
+			config.Status.Ready = false
```
[the remark this answers is not in the document]
Comment (NicoleYson) [reply bootstrap/eks/controllers/eksconfig_controller.go:134]:
```suggestion
```

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 2]
```diff
@@ -133,13 +140,12 @@ func (r *EKSConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr e
 
 	if !cluster.Status.InfrastructureReady {
 		log.Info("Cluster infrastructure is not ready, requeueing")
-		// TODO: set condition
-		return ctrl.Result{}, nil
+		config.Status.Ready = false
```
[the remark this answers is not in the document]
Comment (NicoleYson) [reply bootstrap/eks/controllers/eksconfig_controller.go:143]:
```suggestion
				log.Info("Cluster infrastructure is not ready, requeueing")
		conditions.MarkFalse(config,
			bootstrapv1.DataSecretAvailableCondition,
			bootstrapv1.WaitingForClusterInfrastructureReason,
			clusterv1.ConditionSeverityInfo, "")
		return ctrl.Result{}, nil
```

File: bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go [hunk 3]
```diff
@@ -42,7 +49,8 @@ type EKSConfigTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec EKSConfigTemplateSpec `json:"spec,omitempty"`
+	Spec   EKSConfigTemplateSpec   `json:"spec,omitempty"`
+	Status EKSConfigTemplateStatus `json:"status,omitempty"`
```
Comment (rudoi) [inline bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go:53]: Hmm, this line shouldn't be here. EKSConfigTemplate has no status. Something go wrong with a rebase?

Comment (NicoleYson) [inline bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go:53]:
```suggestion
```

File: bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go [hunk 4]
```diff
@@ -28,6 +28,13 @@ type EKSConfigTemplateSpec struct {
 	Template EKSConfigTemplateResource `json:"template"`
 }
 
+// EKSConfigTemplateStatus defines the observed state of EKSConfigTemplate
+type EKSConfigTemplateStatus struct {
```
Comment (rudoi) [inline bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go:32]: can we get rid of this type entirely? i think it's unused now.

Comment (NicoleYson) [reply bootstrap/eks/api/v1alpha3/eksconfigtemplate_types.go:32]: Removed: https://github.com/newrelic-forks/cluster-api-provider-aws/pull/13/commits/8bd1a585852f7d86e74fca19cccb44a1847c6e8d

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 5]
```diff
@@ -143,17 +152,27 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, scope *EKSConfigSc
...
 
 	if !scope.Cluster.Status.ControlPlaneInitialized {
 		scope.Logger.Info("Cluster has not yet been initialized, requeueing")
-		// TODO? set condition
+		conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
 		return ctrl.Result{}, nil
 	}
 
 	// generate userdata
+	// store userdata as secret (this can basically be totally copied from kubeadm provider - any way to reuse?)
```
Comment (rudoi) [inline bootstrap/eks/controllers/eksconfig_controller.go:169]: nit - i think most, if not all, of these comments can go away

**Pull request merged**